src/backend/aspen/api/views/samples.py

```diff
@@ -194,20 +194,20 @@ async def update_samples(
         for key, value in update_data:
             if key in ["collection_location", "sequencing_date"]:
                 continue
-            if value is not None:  # We need to be able to set private to False!
-                setattr(sample, key, value)
+            setattr(sample, key, value)
         # Location id is handled specially
         if update_data.collection_location:
             loc = await db.get(Location, update_data.collection_location)
             if not loc:
                 raise ex.BadRequestException("location is invalid")
             sample.collection_location = loc
+
         # Sequencing date is handled specially
-        if update_data.sequencing_date:
-            sample.uploaded_pathogen_genome.sequencing_date = (
-                update_data.sequencing_date
-            )
+        sample.uploaded_pathogen_genome.sequencing_date = update_data.sequencing_date  # type: ignore
+        # workaround for our response serializer
         sample.show_private_identifier = True
+
+        sample.generate_public_identifier(already_exists=True)
```

Comment:
Could it be possible that samples with a public identifier don't conform to the format in this function? I find it odd that we have a metadata field for this but then we would always be overwriting it on update here.

Reply:
we would not be overwriting it if the sample has a public identifier, it only autogenerates if the sample has no public_identifier. it's a good callout though because it does look like it overwrites!